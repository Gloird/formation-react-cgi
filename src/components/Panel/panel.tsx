import React, { Component } from "react";

interface Props {
  /**
   * Titre du panel
   */
  titre: string;
}
interface State {}

export default class Panel extends Component<Props, State> {
  state = {};

  render() {
    return (
      <div>
        <span>{this.props.titre}</span>
        {this.props.children}
      </div>
    );
  }
}
