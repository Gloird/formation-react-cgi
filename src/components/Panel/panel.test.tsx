import { render } from "@testing-library/react";
import Panel from "./panel";

describe("Panel", () => {
  test("Vérification de la construction du Panel", () => {
    const renderPanel = render(<Panel titre="Titre du panel" />);

    //Vérifie que un div existe
    expect(renderPanel.container.getElementsByTagName("div")).toHaveLength(1);
    //Vérifie qu'il existe bien un enfant dans le div
    expect(
      renderPanel.container.getElementsByTagName("div").item(0)?.children
    ).toHaveLength(1);
    //Vérifie que l'enfant est bien un SPAN
    expect(
      renderPanel.container
        .getElementsByTagName("div")
        .item(0)
        ?.children.item(0)?.tagName
    ).toBe("SPAN");
  });

  test("Affichage du titre", () => {
    const renderPanel = render(<Panel titre="Titre du panel" />);

    //Titre du panel est il dans le dom
    expect(renderPanel.getByText("Titre du panel")).toBeInTheDocument();
  });
  test("Affichage des children", () => {
    const renderPanel = render(
      <Panel titre="Titre du panel">
        <span data-testid="span1">Span1</span>
        <span data-testid="span2">Span2</span>
      </Panel>
    );
    //Titre du panel est il dans le dom
    expect(renderPanel.getByText("Titre du panel")).toBeInTheDocument();
    //Vérification que le span 1 est afficher (utilisation du testId car les span et div n'ont pas de rôle voir : https://www.w3.org/TR/html-aria/ )
    expect(renderPanel.getByTestId("span1")).toBeInTheDocument();
    //Vérification que le span 2 est afficher (utilisation du testId car les span et div n'ont pas de rôle voir : https://www.w3.org/TR/html-aria/ )
    expect(renderPanel.getByTestId("span2")).toBeInTheDocument();
  });

  test("Snapshot Panel", () => {
    const renderButton = render(<Panel titre="Titre du panel" />);

    //Permet de s'assurer que l'UI ne change pas.
    expect(renderButton.baseElement).toMatchSnapshot();
  });
});
