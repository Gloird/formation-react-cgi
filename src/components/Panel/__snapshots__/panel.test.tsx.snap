// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel Snapshot Panel 1`] = `
<body>
  <div>
    <div>
      <span>
        Titre du panel
      </span>
    </div>
  </div>
</body>
`;
